Web.md

```diff
@@ -124,6 +124,81 @@ MIME type mismatch or `opaque` response types
+
+[StructuredSerializeInternal](https://html.spec.whatwg.org/multipage/infrastructure.html#structuredserializeinternal) adds a new step after step 12:
+
+* Otherwise, if `value` has an [[WebAssembly.Memory]] internal slot, then:
+  1. Let `size` be value.[[BufferObject].
+  2. If !  [IsSharedArrayBuffer](https://tc39.github.io/ecma262/#sec-issharedarraybuffer) is true, then:
+    1. If `forStorage` is true, then throw a
+       ["DataCloneError" DomException](https://heycam.github.io/webidl/#datacloneerror).
+    2. Set `serialized` to { [[Type]]: "WebAssembly.Memory",
+       [[ArrayBufferData]]: value.[[BufferObject]].[[ArrayBufferData],
+       [[ArrayBufferByteLength]]: size,
+       [[AgentCluster]]: the [current Realm Record](https://tc39.github.io/ecma262/#current-realm)'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) }.
+    3. Otherwise:
+       1. Let `dataCopy` be [CreateByteDataBlock](https://tc39.github.io/ecma262/#sec-createbytedatablock)(size).
+          NOTE: This can throw a RangeError exception on allocation failure.
+       2. Perform `CopyDataBlockBytes`(https://tc39.github.io/ecma262/#sec-copydatablockbytes)
+          (dataCopy, 0,
+          value.[[BufferObject]].[[ArrayBufferData]], 0, size).
+       3. Set `serialized` to { [[Type]]: "WebAssembly.Memory",
+          [[ArrayBufferData]]: dataCopy, [[ArrayBufferByteLength]]: size }.
+
+[StructuredDeserialize](https://html.spec.whatwg.org/multipage/infrastructure.html#structureddeserialize) adds a new step after step 12:
+
+* Otherwise, if `serialized`.[[Type]] is "WebAssembly.Memory", then:
+  1. If `targetRealm`'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+     `serialized`.[[AgentCluster]], then throw a
+     ["DataCloneError" DomException](https://heycam.github.io/webidl/#dfn-DOMException).
+  2. Otherwise, set `value` to a new WebAssembly.Memory object in `targetRealm`
+     whose [[ArrayBufferData]] internal slot value is
+     `serialized`.[[ArrayBufferData]]
+     and whose [[ArrayBufferByteLength]] internal
+     slot value is `serialized`.[[ArrayBufferByteLength]].
+
+
+### Structured Clone of a `WebAssembly.Module`
+
+A `WebAssembly.Module` is under some circumstances a
+[cloneable object](https://html.spec.whatwg.org/multipage/infrastructure.html#cloneable-objects), and will impact the
+[HTML Standard](https://html.spec.whatwg.org/multipage/infrastructure.html#serialization-steps) as follows:
+
+[StructuredSerializeInternal](https://html.spec.whatwg.org/multipage/infrastructure.html#structuredserializeinternal) adds a new step after step 12:
+
+* Otherwise, if `value` has an [[WebAssembly.Module]] internal slot, then:
+  1. If `forStorage` is true, implementations MAY chose to exclude
+     "insecure origins", in which case they should throw a
+     ["DataCloneError" DomException](https://heycam.github.io/webidl/#datacloneerror).
```

Comment:
Yes. This needs to decide which object's origin is being tested, per https://github.com/whatwg/html/pull/2518#discussion_r112337423, and needs to link to https://w3c.github.io/webappsec-secure-contexts/#settings-object applied to one of the settings objects involved (depending on which one you choose).